java/client/src/main/java/glide/api/commands/StreamBaseCommands.java

```diff
@@ -1372,4 +1372,270 @@ CompletableFuture<GlideString[]> xclaimJustId(
      */
     CompletableFuture<Map<GlideString, Object>[]> xinfoConsumers(
             GlideString key, GlideString groupName);
+
+    /**
+     * Returns information about the stream stored at key <code>key</code>.<br>
+     * To get more detailed information use {@link #xinfoStreamFull(String)} or {@link
+     * #xinfoStreamFull(String, int)}.
+     *
+     * @see <a href="https://valkey.io/commands/xinfo-stream/">valkey.io</a> for details.
+     * @param key The key of the stream.
+     * @return A <code>Map</code> of stream information for the given <code>key</code>. See the
+     *     example for a sample response.
+     * @example
+     *     <pre>{@code
+     * // example of using the API:
+     * Map<String, Object> response = client.xinfoStream("myStream").get();
+     * // the response contains data in the following format:
+     * Map<String, Object> data = Map.of(
+     *     "length", 4L,
+     *     "radix-tree-keys", 1L,
+     *     "radix-tree-nodes", 2L,
+     *     "last-generated-id", "1719877599564-0",
+     *     "max-deleted-entry-id", "0-0",
+     *     "entries-added", 4L,
+     *     "recorded-first-entry-id", "1719710679916-0",
+     *     "groups", 1L,
+     *     "first-entry", Map.of(
```

Comment:
Instead of a map, this is actually a list, see my previous comment in BaseClient.java, which also includes links to the Python example, which can be used as a reference for what the return value looks like